Pre-RoboCup 2020 season version

alex-mitrevski opened the pull request:
This PR introduces the following features and changes:

* An action for opening handles (https://github.com/b-it-bots/mas_domestic_robotics/pull/184)
* An action for pushing/pulling objects (https://github.com/b-it-bots/mas_domestic_robotics/pull/210)
* A head controller abstract interface (https://github.com/b-it-bots/mas_domestic_robotics/pull/209); an instance of the controller was added in the `perceive_plane` action (https://github.com/b-it-bots/mas_domestic_robotics/pull/213) and is used for basic recovery there (https://github.com/b-it-bots/mas_domestic_robotics/pull/215)
* Updates to the `storing_groceries` PDDL domain and default planner change (LAMA is used instead of POPF) (https://github.com/b-it-bots/mas_domestic_robotics/pull/206)

Various fixes are also included:

* DMP interface refactoring in `mdr_move_arm_action` (https://github.com/b-it-bots/mas_domestic_robotics/pull/205)
* Bug fixes in the perceive plane action (https://github.com/b-it-bots/mas_domestic_robotics/pull/211)
* Fixed usage of the `Person` message (https://github.com/b-it-bots/mas_domestic_robotics/pull/212)
* Bug fixes in the `place` action client (https://github.com/b-it-bots/mas_domestic_robotics/pull/216)
* Bug fixes in the `pickup` action (https://github.com/b-it-bots/mas_domestic_robotics/pull/200)
* Launch file fixes in the `listen` action (https://github.com/b-it-bots/mas_domestic_robotics/pull/195)

## TODOs before merging

* [x] Remove the object grasping functionality from the `push_pull_object` action and instead introduce a parameter in the `pickup` action that allows the arm to remain at the grasping position; this will allow chaining manipulation actions (PRs https://github.com/b-it-bots/mas_domestic_robotics/pull/222 and https://github.com/b-it-bots/mas_domestic_robotics/pull/223)

alex-mitrevski committed:
[handle_open_action] Added skeleton code for the action implementation

af-a committed:
[handle_open_action] (test commit) Add basic SM sequence code

af-a committed:
[handle_open_action] Add basic action server definitions

af-a committed:
[handle_open_action] Add initial action execution logic

af-a committed:
[handle_open_action] Add test client goal specification

af-a committed:
[handle_open_action] Add launch file and server parameters

af-a committed:
[handle_open_action] Correct action parameter initializations

af-a committed:
[handle_open_action] Add pre-grasp wrist rotation to action

af-a committed:
[handle_open_action] Add sampling and saving of random grasp poses

af-a committed:
[handle_open_action] Modify action's move backward distance

af-a committed:
[handle_open_action] Add uniformly sampled grasp poses

af-a committed:
[handle_open_action] Incorporate sampling from learned Gaussian dist

af-a committed:
[handle_open_action] Update client file to include latest changes

af-a committed:
[handle_open_action] Add grasp poses sampled from the learned dist

argenos committed:
mdr_speech_matching: Fix fuzzywuzzy dependency

af-a committed:
[handle_open_action] Add vision-based failure detection (in progress)

argenos committed:
readme: Add badges for license and version

argenos committed:
codeowners: Update reviewers to teams for pullpanda

argenos committed:
codeowners: Add pullpanda's proxy teams to reduce notifications

robertocaiwu committed:
[speech_recognition] point to kaldi models in  mas_models

PatrickNa committed:
Merge pull request #195 from robertocaiwu/hotfix/refactor_speech_model_directory

[speech_recognition] point to kaldi models in mas_models

sthoduka committed:
[move_arm_action/dmp] fix parameter name sigma_threshold_lower

sthoduka committed:
[pickup_action] add missing import of PickupGoal

af-a committed:
[handle_open_action] Add force-based failure detection (testing)

af-a committed:
[hand_over_action] Clean up code

njanirudh committed:
[handle_open_action] Add code for peception based failure detection

af-a committed:
[handle_open_action] Make fixes to pass Codacy check

alex-mitrevski committed:
[move_arm/dmp] Code cleanup and documentation

alex-mitrevski committed:
[move_arm/dmp] Fixed incomplete docstrings

alex-mitrevski committed:
[handle_open_action/package.xml] Added package authors

alex-mitrevski committed:
[handle_open_action] Renamed the launcher and removed the HSR-specific parameters (they've been moved to a launcher in mas_hsr)

alex-mitrevski committed:
[handle_open_action] Removed HSR-specific values as default params and linter fixes

alex-mitrevski committed:
[handle_open_action] Removed the init_end_effector_pose parameter from the action goal definition

alex-mitrevski committed:
[handle_open_action] Added a fn for aligning the base with the handle pose (taken from the pickup action)

alex-mitrevski committed:
[handle_open_action/README] Added a short description of the action, its parameters, and the dependencies

alex-mitrevski committed:
[handle_open_action/client_test] Fixed a syntax error (unindented block)

alex-mitrevski committed:
[handle_open_action] Moved the pose files to a 'learning_data' dir and wrote a short description in the README

alex-mitrevski committed:
[store_groceries] Updated the domain to allow for a simple goal specification (a single groceries_stored predicate after perceiving)

alex-mitrevski committed:
[store_groceries/sample_initial_problem] Added some objects to the environment

alex-mitrevski committed:
[rosplan_interface] Changed the planner to Metric-FF

alex-mitrevski committed:
[planning_behaviours/dispatch_plan] Corrected function call

alex-mitrevski committed:
[store_groceries/domain] Added a context parameter to the pick and place action

alex-mitrevski committed:
[store_groceries] Added a state for managing plan goals and an SM that instantiates the state

alex-mitrevski committed:
[store_groceries] Reintroduced object categories in the domain

alex-mitrevski committed:
[store_groceries] Added a sample pddl problem file

alex-mitrevski committed:
[handle_open_action/learning_data] Added labels

alex-mitrevski committed:
[handle_open_action] Added a DMP for the action (same as the placing dmp)

alex-mitrevski committed:
[handle_open_action/action_states] Cleanup and minor docs changes

alex-mitrevski committed:
Merge branch 'devel' of github.com:b-it-bots/mas_domestic_robotics into feature/handle-open-action

alex-mitrevski committed:
Merge pull request #184 from b-it-bots/feature/handle-open-action

Handle opening action

alex-mitrevski committed:
[move_arm_action/dmp] Fixed a bug in the WBM velocity split

Note: I'd introduced the bug while refactoring the component

alex-mitrevski committed:
[perception] Added a base head controller

alex-mitrevski committed:
Added an action for pushing/pulling objects to a goal location

alex-mitrevski committed:
[push_pull_object_action] Removed an old leftover docstring and resolved a default arg warning

alex-mitrevski committed:
[perceive_plane_action] import ObjectDetector -> import PlaneDetector

This change is required due to the recent refactoring of mas_perception_libs (PR https://github.com/b-it-bots/mas_perception_libs/pull/15)

alex-mitrevski committed:
[find_people] Cleaned up the code and adapted the person msg to the new format

The person message format was modified with PR https://github.com/b-it-bots/mas_perception_msgs/pull/6

alex-mitrevski committed:
[find_my_mates/move_to_person] Adapted the person msg to the new format

A utility function for setting the pose of the person to a safe pose
(which is one meter away) was moved here from the find_person action

alex-mitrevski committed:
[find_my_mates/move_to_person] Fixed util fn name

alex-mitrevski committed:
[perceive_plane] Added a head ctrlr instance to the action

alex-mitrevski committed:
[planner_interface] Fixed the type of the dispatch service client

The service used to be of type std_msgs.Empty, but the type seems to
have been changed

alex-mitrevski committed:
[store_groceries/domain.pddl] Removed the 'fluents' requirement

alex-mitrevski committed:
[perceive_plane_client] Parameter name update in the object_category predicate

cat -> class (this is in accordance with the changes to the PDDL domain)

alex-mitrevski committed:
[store_groceries/goal_selector] Fixed goal selection condition

alex-mitrevski committed:
[store_groceries/goal_selector] Added a fn for setting the object storing locations before planning to store the groceries

alex-mitrevski committed:
[store_groceries/domain.pddl] Modified the domain so that it's without conditional effects

This is due to the fact that ROSPlan doesn't seem to support conditional
effects and forall goals: https://github.com/KCL-Planning/ROSPlan/issues/178

alex-mitrevski committed:
[store_groceries/problem.pddl] Updated the sample problem

alex-mitrevski committed:
[store_groceries/goal_selector] Modifications to handle the updated simplified domain

alex-mitrevski committed:
[rosplan.launch] Changed the planner to LAMA

alex-mitrevski committed:
[rosplan.launch] Modified the LAMA planner command

This is to resolve issue https://github.com/b-it-bots/ROSPlan/issues/5

divindevaiah committed:
[perceive_plane] added simple recovery (issue #214)

alex-mitrevski committed:
[head_controller_base] Added an actions field that lists the possible head movements

alex-mitrevski committed:
[perceive_plane/action_states/recovering] Selecting an action from the actions field of the head ctrlr

alex-mitrevski committed:
[place/throw_client] Action name change to match the names in the action ontology

...since the names in the ontology start with a capital letter

minhnh reviewed (commented):
Is the [`weights_table_grasp.yaml` file](https://github.com/b-it-bots/mas_domestic_robotics/tree/devel/mdr_planning/mdr_actions/mdr_manipulation_actions/mdr_push_pull_object_action/config/trajectory_weights/weights_table_grasp.yaml) the same with all the arm moving actions? If so then this file may be removed.

Otherwise PR looks fine to me

moriarty commented:
BTW- it’s awesome that you guys have SonarCloud working in the repo!

alex-mitrevski committed:
[pickup_action] Added context to the goal

In the current implementation, this prevents moving the arm back to a
safe configuration if the context is tabletop manipulation

alex-mitrevski committed:
[pickup_action] Updated the client to retrieve the context and set it in the goal

alex-mitrevski committed:
[push_pull_action] Added context to the goal and removed grasping workflow

alex-mitrevski committed:
[manipulation_behaviours/pick_closest] Added a grasping context argument

alex-mitrevski committed:
[push_pull] Removed the grasping parameters from the README and server script

alex-mitrevski committed:
[push_pull_action] Removed unused import
